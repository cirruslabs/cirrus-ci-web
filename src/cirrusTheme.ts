import { atom, selector } from 'recoil';

import { grey, lightGreen, orange, red } from '@mui/material/colors';
import { ThemeOptions } from '@mui/material/styles';

import { localStorageEffect } from 'utils/recoil';

export const prefersDarkModeState = atom({
  key: 'CurrentlyPrefersDarkMode',
  default: false,
  effects_UNSTABLE: [localStorageEffect('CurrentlyPreferredTheme')],
});

export const muiThemeOptions = selector({
  key: 'muiThemeOptions',
  get: ({ get }) => {
    const prefersDarkMode = get(prefersDarkModeState);
    return prefersDarkMode ? muiDarkTheme : muiLightTheme;
  },
});

export const cirrusColorsState = selector({
  key: 'cirrusColorsState',
  get: ({ get }) => {
    const prefersDarkMode = get(prefersDarkModeState);
    return {
      undefined: prefersDarkMode ? grey['600'] : grey['400'],
    };
  },
});

export const cirrusThemeOptions = selector({
  key: 'cirrusThemeOptions',
  get: ({ get }) => {
    const prefersDarkMode = get(prefersDarkModeState);
    return prefersDarkMode ? cirrusDarkTheme : cirrusLightTheme;
  },
});

let cirrusBaseTheme: ThemeOptions = {
  typography: {
    fontFamily: 'Roboto, sans-serif',
  },
  shape: {
    borderRadius: 2,
  },
};

export let cirrusLightTheme: ThemeOptions = {
  ...cirrusBaseTheme,
  palette: {
    mode: 'light',
    primary: {
      main: grey['900'],
      dark: grey['900'],
      light: grey['50'],
      contrastText: grey['50'],
    },
    secondary: {
      main: grey['700'],
      dark: grey['700'],
      light: grey['50'],
      contrastText: grey['50'],
    },
    action: {
      hover: grey['200'],
    },
    success: {
      light: lightGreen['300'],
      main: lightGreen['500'],
      dark: lightGreen['700'],
    },
    error: {
      light: red['300'],
      main: red['500'],
      dark: red['700'],
    },
    warning: {
      light: orange['300'],
      main: orange['500'],
      dark: orange['700'],
    },
  },
  components: {
    MuiChip: {
      styleOverrides: {
        avatar: {
          marginLeft: 0,
          marginRight: 0,
          height: 32,
          width: 32,
        },
      },
    },
    MuiSkeleton: {
      styleOverrides: {
        root: {
          backgroundColor: grey['800'],
        },
      },
    },
    MuiTableCell: {
      styleOverrides: {
        body: {
          padding: 6, // it was changed from v4
        },
      },
    },
  },
};

export let cirrusDarkTheme: ThemeOptions = {
  ...cirrusBaseTheme,
  palette: {
    mode: 'dark',
    primary: {
      main: grey['900'],
      dark: grey['900'],
      light: grey['50'],
      contrastText: grey['50'],
    },
    secondary: {
      main: grey['700'],
      dark: grey['700'],
      light: grey['50'],
      contrastText: grey['50'],
    },
    success: {
      light: lightGreen['400'],
      main: lightGreen['600'],
      dark: lightGreen['800'],
    },
    error: {
      light: red['400'],
      main: red['600'],
      dark: red['800'],
    },
    warning: {
      light: orange['400'],
      main: orange['600'],
      dark: orange['800'],
    },
  },
  components: {
    MuiChip: {
      styleOverrides: {
        avatar: {
          marginLeft: 0,
          marginRight: 0,
          height: 32,
          width: 32,
        },
      },
    },
    MuiTableCell: {
      styleOverrides: {
        body: {
          padding: 6, // it was changed from v4
        },
      },
    },
    MuiTab: {
      styleOverrides: {
        root: {
          '&.Mui-selected': {
            color: grey['50'],
          },
        },
      },
    },
  },
};

export let muiLightTheme: ThemeOptions = {
  palette: cirrusLightTheme.palette,
};

export let muiDarkTheme: ThemeOptions = {
  palette: cirrusDarkTheme.palette,
};

export const cirrusOpenDrawerState = atom({
  key: 'CirrusOpenDrawer',
  default: false,
  effects_UNSTABLE: [localStorageEffect('CirrusOpenDrawer')],
});
