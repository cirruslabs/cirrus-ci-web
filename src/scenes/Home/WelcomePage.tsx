import React from 'react';
import Paper from '@mui/material/Paper';
import ReactMarkdown from 'react-markdown';
import withStyles from '@mui/styles/withStyles';
import { Helmet as Head } from 'react-helmet';

const styles = theme => ({
  paper: {
    padding: theme.spacing(3.0),
    margin: theme.spacing(1.0),
  },
});

let welcomePage = props => (
  <Paper className={props.classes.paper}>
    <Head>
      <title>Welcome - Cirrus CI</title>
    </Head>
    <ReactMarkdown># Welcome to Cirrus CI.</ReactMarkdown>
    <ReactMarkdown>
      Please [**sign in**](https://api.cirrus-ci.com/redirect/auth/github) to see your recent builds.
    </ReactMarkdown>
    <ReactMarkdown>Just got here? Visit [**our documentation**](https://cirrus-ci.org).</ReactMarkdown>
  </Paper>
);

export default withStyles(styles)(welcomePage);
