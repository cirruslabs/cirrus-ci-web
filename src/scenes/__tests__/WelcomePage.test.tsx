import React from 'react';
import WelcomePage from '../Home/WelcomePage';
import renderer from 'react-test-renderer';
import { ThemeProvider } from '@mui/styles';
import { createTheme } from '@mui/material/styles';
import { cirrusDarkTheme } from '../../cirrusTheme';

it('renders correctly', () => {
  let theme = createTheme(cirrusDarkTheme);
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <WelcomePage />
      </ThemeProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
