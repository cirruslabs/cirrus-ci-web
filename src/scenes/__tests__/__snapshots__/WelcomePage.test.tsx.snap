// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 welcomePage-paper-1 css-1ps6pg7-MuiPaper-root"
>
  <h1>
    Welcome to Cirrus CI.
  </h1>
  <p>
    Please 
    <a
      href="https://api.cirrus-ci.com/redirect/auth/github"
    >
      <strong>
        sign in
      </strong>
    </a>
     to see your recent builds.
  </p>
  <p>
    Just got here? Visit 
    <a
      href="https://cirrus-ci.org"
    >
      <strong>
        our documentation
      </strong>
    </a>
    .
  </p>
</div>
`;
